processing/src/main/java/org/apache/druid/query/Query.java

```diff
@@ -271,4 +275,24 @@ default Set<String> getRequiredColumns()
   {
     return null;
   }
+
+  default CursorBuildSpec asCursorBuildSpec(@Nullable QueryMetrics<?> queryMetrics)
+  {
+    final List<Interval> intervals = getIntervals();
+    if (intervals.size() > 1) {
```

Comment:
should be `!= 1`, due to the `Iterables.getOnlyElement` later.

BTW, the error message and only-element retrieval can be combined, if you like, using `CollectionUtils.getOnlyElement` (which also takes an error supplier).

Reply:
cool, i didn't know about this method, switched to use it